config_parser_test.cc

```diff
@@ -1,11 +1,173 @@
 #include "gtest/gtest.h"
 #include "config_parser.h"
 
-TEST(NginxConfigParserTest, SimpleConfig) {
+/*****************************************************************************
+    BASIC TESTS
+*****************************************************************************/
+
+// Tests basic config with a single statement
+TEST(NginxConfigParserTest, ParseSimpleConfig) {
   NginxConfigParser parser;
   NginxConfig out_config;
+  EXPECT_TRUE(parser.Parse("example_config", &out_config));
+}
+
+// Tests ToString method on simple config
+TEST(NginxConfigParserTest, ToStringSimple) {
+	NginxConfigStatement statement;
+	statement.tokens_.push_back("foo");
+	statement.tokens_.push_back("bar");
+	EXPECT_EQ(statement.ToString(0), "foo bar;\n");
+}
+
+// Tests ToString method on config with a block
+TEST(NginxConfigParserTest, ToStringBlock) {
+	NginxConfigStatement statement;
+	statement.tokens_.push_back("server");
+	statement.tokens_.push_back("{");
+	statement.tokens_.push_back("listen");
+	statement.tokens_.push_back("80");
+	statement.tokens_.push_back(";");
+	statement.tokens_.push_back("}");
+	EXPECT_EQ(statement.ToString(0), "server { listen 80 ; };\n");
+}
+
+/*****************************************************************************
+    PARSING STRINGS
+*****************************************************************************/
+
+// Test fixture for parsing strings
+class NginxStringConfigTest : public ::testing::Test {
+protected:
+	bool ParseString(const std::string config_string) {
+		std::stringstream config_stream(config_string);
+		return parser_.Parse(&config_stream, &out_config_);
+	}
+	NginxConfigParser parser_;
+	NginxConfig out_config_;
+};
+
+// Verify config parser correctly separates a simple statement into tokens
+TEST_F(NginxStringConfigTest, SimpleConfigValid) {
+	EXPECT_TRUE(ParseString("foo bar;"));
+	EXPECT_EQ(out_config_.statements_.size(), 1); // config has one statement
+	EXPECT_EQ(out_config_.statements_.at(0)->tokens_.size(), 2);
+	EXPECT_EQ(out_config_.statements_.at(0)->tokens_.at(0), "foo");
+	EXPECT_EQ(out_config_.statements_.at(0)->tokens_.at(1), "bar");
+	EXPECT_EQ(out_config_.statements_.at(0)->child_block_,
+		std::unique_ptr<NginxConfig>()); // child block is empty
+}
 
-  bool success = parser.Parse("example_config", &out_config);
+// Statements must end in semicolons
+TEST_F(NginxStringConfigTest, SimpleConfigInvalid) {
+	EXPECT_FALSE(ParseString("foo bar"));
+}
+
+// Blocks with matching { and } brackets are valid
+TEST_F(NginxStringConfigTest, BlockConfigValid) {
```

Comment:
Nice testing. It's good that you tested the full contents of child_block, it also helps to see how the config parser works (sort of like documentation).